cmd/monaco/download/download_configs_test.go

```diff
@@ -281,6 +293,7 @@ func TestDownload_Options(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			t.Setenv(featureflags.Temporary[featureflags.Segments].EnvName(), "true")
```

Comment:
I would argue against that. All cases need to pass in when FF is on. E.g.  OpenPipeline  also have the FF to deactivate them. But it's default is on.

Reply:
In my opinion, not so optimal designed test as it could have side effects on other test cases in the `TestDownload`. What do you think @Laubi ?

Reply:
I don't understand this part about side effects on other test cases.

Reply:
To my understanding, side-effects that are not possible to test here are:
- Download all (FF for segments is disabled; segments should not be downloaded, but in your case, this will be true in every test case)
- Download segments only, but with FF disabled (get an error message...)

Reply:
I agree that setting the FF inside the test should not be done, also for one very important reason: We will merge this PR, and might release a new version of Monaco before we enable the FF. I argue that the behavior should be tested of the off- and on-state of the F so that we can be assured that it always works as intended.

Reply:
Hehe...
That confusion happens when under the test (SUT) is not exported, but rather non-exported function :smile: 

By this point, if FF is disabled (by default like for segments, or by FF env var like for open-pipeline) it's going to be false. 
And FFs are tested in downloadopts_test.go.

I agree that the code and test style needs improvement.

Reply:
discussed in the call. Changes made